IGNITE-22384 Replace Conan package manager with direct dependency management using cmake (#3851)
apache/ignite-3: modules/platforms/cpp/DEVNOTES.md

### Prerequisites

* C++ compiler supporting C++ 17
* One of the build systems: make, ninja, MS Visual Studio, etc.
* CMake 3.10+

To build the ODBC driver, it is required to have an ODBC driver manager with headers on your system. On Windows, it 
comes with your OS. On Unix-like systems, you may need to install one; for example, unixODBC.

### CMake Options and Typical Build Configurations

The CMake project supports multiple configuration options that can be used to define exactly what and how should be 
built. You can check the list of available CMake options with the `cmake -LAH` command. 
Following are the project-specific options:

> * USE_LOCAL_DEPS={ON|OFF} - OFF by default. Indicates that build system will use locally installed dependencies.
> libraries and will try to download dependencies only if it will not be found locally.

Reviewer: ```suggestion
* USE_LOCAL_DEPS={ON|OFF} - OFF by default. Indicates that build system will use locally installed dependencies,
 and will try to download them only if they can not be found locally.
```
Also, I think we should not try to download anything if we can not find it. Just give user an error.
